### PR TITLE
Fix silent publish splash widget not appearing in screen center

### PR DESCRIPTION
**Pull request by mruegenberg**
Without this fix the widget appears at screen top left for me (PySide2, Nuke 13)

### CODE REVIEW

**Comment by mottosso** [comment]
This will be so heavily dependent on OS and window manager within the OS, and whether you have resolution scaling (e.g. a retina screen) or not. If you can limit this to being applied to your environment, e.g. Gnome 3 with Wayland or Windows 11, it would be safe to merge.

**Comment by mruegenberg** [comment]
Hmm, I'd think that the calculation would take retina  into account and also seems like this code works for a lot of other people from where I found it...
Right now the widget just shows up on screen top left, which feels like a bug, compared also to screenshots?

Do you have a recommendation on how I would go about limiting this to a specific environment?

**Comment by mottosso** [comment]
> Do you have a recommendation on how I would go about limiting this to a specific environment?

Yes, put it behind this:

```py
import sys

if sys.platform == "darwin":
   # do_mac_things
```

**Comment by mruegenberg** [comment]
Okay, in my case it would be Linux, and I'll happily add a check for that.

How specific would you want the code that checks this to be to merge this? To isolate my specific desktop environment (CentOS 7 + MATE) and whether using X11 or Wayland I could probably use some environment variables...?

**Comment by mruegenberg** [comment]
Hey happy new year!
I've now added a check that limits this change to both Linux and X11 as I don't have any Wayland system available, so I can't test it.

File: pyblish_nuke/lib.py [hunk 1]
```diff
@@ -176,6 +176,13 @@ def __init__(self, parent=None):
         # Center widget on screen
         self.resize(100, 100)
 
+        # code was so far only tested on Linux/X11 with 1 and 2 screens
+        if sys.platform == "linux" and os.environ['XDG_SESSION_TYPE'] == "x11":
```
[the remark this answers is not in the document]
Comment (mruegenberg) [reply pyblish_nuke/lib.py]: Oh yes, good point. I've corrected that now.

**Comment by mottosso** [comment]
Looks good, nice work.

**Pull request merged**